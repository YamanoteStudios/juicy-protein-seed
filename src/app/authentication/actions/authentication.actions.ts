import { Action } from '@ngrx/store';
import { Credentials } from '../models/credentials.model';
import { UserModel } from '../models/user.model';

export enum CredentialsActionTypes {
  Login = '[Login Page] Login',
  LoginSuccess = '[Auth API] Login Success',
  Logout = '[Authentication] Confirm Logout',

}

export class Login implements Action {
  readonly type = CredentialsActionTypes.Login;

  constructor(public payload: Credentials) {}
}


export class Logout implements Action {
    readonly type = CredentialsActionTypes.Logout;
  }



export type CredentialsActions =
| Login
| Logout;

  