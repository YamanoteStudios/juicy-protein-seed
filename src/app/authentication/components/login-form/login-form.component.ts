import { Component, OnInit, EventEmitter, Input, Output } from '@angular/core';
import { FormGroup, FormControl, Validators } from '@angular/forms';

import { Credentials } from '../../models/credentials.model';




@Component({
  selector: 'newsn-login-form',
  templateUrl: './login-form.component.html',
  styleUrls: ['./login-form.component.css']
})
export class LoginFormComponent implements OnInit {
  @Input() error: string | null;

  @Input() set disabled(isDisabled: boolean) {
    if (isDisabled) {
      this.loginForm.disable();
    } else {
      this.loginForm.enable();
    }
  };

  @Output() submitted = new EventEmitter<Credentials>();

  loginForm = new FormGroup({
    username: new FormControl('', [Validators.required]),
    password: new FormControl('', [Validators.required])
  });

  ngOnInit() {}

  submit() {
    const value: Credentials = this.loginForm.value;

    if (this.loginForm.valid) {
      this.submitted.emit(value);
    }
  }
}
