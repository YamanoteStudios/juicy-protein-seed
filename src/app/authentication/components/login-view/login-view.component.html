<newsn-login-form></newsn-login-form>
   