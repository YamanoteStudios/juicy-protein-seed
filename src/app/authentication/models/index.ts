export * from './config.model';
export * from './credentials.model';
export * from './member.model';
export * from './user.model';