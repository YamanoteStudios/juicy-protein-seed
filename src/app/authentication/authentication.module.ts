// Native angular modules
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import {
  MatInputModule,
  MatButtonModule,
  MatCardModule,
  MatDialogModule,
} from '@angular/material';
import { EffectsModule } from '@ngrx/effects';


// Custom Modules & Components 
import { RouteAuthenticationModule } from './route-authentication.module';
import { LoginViewComponent } from './components/login-view/login-view.component';
import { LoginFormComponent } from './components/login-form/login-form.component';
import { UserHomeComponent } from './components/user-home/user-home.component';
import { LogoutPromptComponent } from './components/logout-prompt/logout-prompt.component';
import { AuthenticationEffects } from '../authentication/effects/authentication.effects';


const NATIVE_COMPONENTS = [];
const NATIVE_MODULES = [
  NgModule,
  CommonModule,
  ReactiveFormsModule,
  MatInputModule,
  MatButtonModule,
  MatCardModule,
  MatDialogModule,
  EffectsModule,
];


const CUSTOM_MODULES = [
  RouteAuthenticationModule
];

const CUSTOM_COMPONENTS = [
  LoginViewComponent,
  UserHomeComponent,
  LogoutPromptComponent,
  LoginFormComponent,
]


@NgModule({
    imports: [
      CommonModule,
      MatButtonModule,
      MatCardModule,
      MatDialogModule,
      MatInputModule,
      ReactiveFormsModule,
      EffectsModule.forFeature([AuthenticationEffects]),
    
    ],
    declarations: [
      ...CUSTOM_COMPONENTS
      
    ],
    entryComponents: [LogoutPromptComponent],
  })
  export class AuthenticationModule {}
  