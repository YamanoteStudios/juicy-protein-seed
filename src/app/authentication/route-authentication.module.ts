
// Native Components
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';



// Custom Components
import { LoginViewComponent } from './components/login-view/login-view.component';
import { UserHomeComponent } from './components/user-home/user-home.component';
import {
  AuthenticationGuardService
} from './services/authentication-guard.service';

@NgModule({
  imports: [
    RouterModule.forChild([
      { path: 'login', component: LoginViewComponent },
      {
        path: 'home',
        component: UserHomeComponent,
        canActivate: [AuthenticationGuardService],
      }

    ]),
  ],
  providers: [AuthenticationGuardService]
})
export class RouteAuthenticationModule { }
