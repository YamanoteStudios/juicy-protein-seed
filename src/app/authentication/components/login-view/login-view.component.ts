import { Component, OnInit} from '@angular/core';
import { Store } from '@ngrx/store';
import * as fromStore from 'src/app/state';

// Grab the Logout Actions
import { Login } from '../../actions/authentication.actions';

//Grab Credentials Models 
import {Credentials} from '../../models/credentials.model';

@Component({
  selector: 'newsn-login-view',
  templateUrl: './login-view.component.html',
  styleUrls: ['./login-view.component.css']
})
export class LoginViewComponent implements OnInit {
  error$ = this.store.select(fromStore.selectLoginViewError);
  pending$ = this.store.select(fromStore.selectLoginViewPending);


  constructor(private store: Store<fromStore.State>) {}

  ngOnInit() {
  }

  onLogin(credentials: Credentials) {
    this.store.dispatch(new Login(credentials));
  }
}
