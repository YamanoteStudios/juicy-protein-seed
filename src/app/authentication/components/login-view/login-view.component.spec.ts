import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { createComponentFixture, provideMagicalMock } from 'angular-testing-library';
import { Store } from '@ngrx/store';
import { BehaviorSubject } from 'rxjs';

import { LoginFormComponent } from '../login-form/login-form.component';
import { LoginViewComponent } from '../login-view/login-view.component';
import { Login } from '../../actions/authentication.actions';

describe('Login View Component', () => {
  const fixture = createComponentFixture({
    component: LoginViewComponent,
    declarations: [LoginFormComponent],
    providers: [
      provideMagicalMock(Store)
    ]
  });
  
  const error$ = new BehaviorSubject<string>('');
  const pending$ = new BehaviorSubject<boolean>(false);
  let store: Store<any>;

  beforeEach(async() => {
    await fixture.compile({
      error$,
      pending$
    });

    store = fixture.get(Store);    
  });

  it('should compile', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch an action when onLogin is called', () => {
    const credentials = {
      username: 'test',
      password: 'test'
    };

    fixture.instance.onLogin(credentials);

    expect(store.dispatch).toHaveBeenCalledWith(new Login(credentials));
  });
});
