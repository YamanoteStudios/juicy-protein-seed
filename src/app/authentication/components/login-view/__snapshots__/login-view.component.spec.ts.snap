// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View Component should compile 1`] = `
<newsn-login-view
  error$={[Function BehaviorSubject]}
  pending$={[Function BehaviorSubject]}
  store={[Function Object]}
>
  <newsn-login-form
    ng-reflect-disabled="false"
    ng-reflect-error=""
  >
    <mat-card>
      <mat-card-title>
         Login 
      </mat-card-title>
      <mat-card-content>
        <form>
          <mat-form-field>
            <input
              formcontrolname="username"
              matinput=""
              placeholder="Username"
              type="text"
            />
            
            <mat-error>
              Username is required
            </mat-error>
          </mat-form-field>
          <mat-form-field>
            <input
              formcontrolname="password"
              matinput=""
              placeholder="Password"
              type="password"
            />
            
            <mat-error>
              Password is required
            </mat-error>
          </mat-form-field>
          
          <div
            class="loginButtons"
          >
            <button
              mat-button=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </mat-card-content>
    </mat-card>
  </newsn-login-form>
</newsn-login-view>
`;
